sklearn/model_selection/_split.py

```diff
@@ -1472,6 +1472,11 @@ def _iter_indices(self, X, y, groups=None):
         y = check_array(y, ensure_2d=False, dtype=None)
         n_train, n_test = _validate_shuffle_split(n_samples, self.test_size,
                                                   self.train_size)
+
+        if y.ndim == 2:
+            # for multi-label y, map each distinct row to a unique identifier:
+            y = np.array([hash(tuple(row)) for row in y])
```

Comment:
Uniformity certainly doesn't matter here, but uniqueness would.

However, these should all be tuples of integers, which I hope python handles well. Another alternative would be `[str(row) for row in y]` which should guarantee the desired behaviour, what do you think?